xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/Timers.h

```diff
@@ -0,0 +1,816 @@
+  /// | **Recording group** | `unsigned int` | @ref PVRTimer::SetRecordingGroup "SetRecordingGroup" | @ref PVRTimer::GetRecordingGroup "GetRecordingGroup" | *optional*
+  /// | **First start day** | `time_t` | @ref PVRTimer::SetFirstDay "SetFirstDay" | @ref PVRTimer::GetFirstDay "GetFirstDay" | *optional*
+  /// | **Used timer weekdays** | `unsigned int` | @ref PVRTimer::SetWeekdays "SetWeekdays" | @ref PVRTimer::GetWeekdays "GetWeekdays" | *optional*
+  /// | **Prevent duplicate episodes** | `unsigned int` | @ref PVRTimer::SetPreventDuplicateEpisodes "SetPreventDuplicateEpisodes" | @ref PVRTimer::GetPreventDuplicateEpisodes "GetPreventDuplicateEpisodes" | *optional*
+  /// | **EPG unique identifier** | `unsigned int` | @ref PVRTimer::SetEPGUid "SetEPGUid" | @ref PVRTimer::GetEPGUid "GetEPGUid" | *optional*
+  /// | **Margin start** | `unsigned int` | @ref PVRTimer::SetMarginStart "SetMarginStart" | @ref PVRTimer::GetMarginStart "GetMarginStart" | *optional*
+  /// | **Margin end** | `unsigned int` | @ref PVRTimer::SetMarginEnd "SetMarginEnd" | @ref PVRTimer::GetMarginEnd "GetMarginEnd" | *optional*
+  /// | **Genre type** | `int` | @ref PVRTimer::SetGenreType "SetGenreType" | @ref PVRTimer::GetGenreType "GetGenreType" | *optional*
+  /// | **Genre sub type** | `int` | @ref PVRTimer::SetGenreSubType "SetGenreSubType" | @ref PVRTimer::GetGenreSubType "GetGenreSubType" | *optional*
+  /// | **Series link** | `std::string` | @ref PVRTimer::SetSeriesLink "SetSeriesLink" | @ref PVRTimer::GetSeriesLink "GetSeriesLink" | *optional*
+
+  /// @addtogroup cpp_kodi_addon_pvr_Defs_Timer_PVRTimer
+  /// @copydetails cpp_kodi_addon_pvr_Defs_Timer_PVRTimer_Help
+  //@{
+
+  /// @brief **required**\n
+  /// The index of this timer given by the client.
+  ///
+  /// @ref PVR_TIMER_NO_CLIENT_INDEX indicates that the index was not yet set
+  /// by the client, for example for new timers created by Kodi and passed the
+  /// first time to the client. A valid index must be greater than
+  /// @ref PVR_TIMER_NO_CLIENT_INDEX.
+  ///
+  void SetClientIndex(unsigned int clientIndex) { m_cStructure->iClientIndex = clientIndex; }
+
+  /// @brief To get with @ref SetClientIndex changed values.
+  unsigned int GetClientIndex() const { return m_cStructure->iClientIndex; }
+
+  /// @brief **required**\n
+  /// The state of this timer.
+  ///
+  /// @note @ref PVR_TIMER_STATE_NEW is default.
+  ///
+  ///
+  /// --------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// kodi::addon::PVRTimer tag;
+  /// tag.SetState(PVR_TIMER_STATE_RECORDING);
+  /// ~~~~~~~~~~~~~
+  ///
+  void SetState(PVR_TIMER_STATE state) { m_cStructure->state = state; }
+
+  /// @brief To get with @ref SetState changed values.
+  PVR_TIMER_STATE GetState() const { return m_cStructure->state; }
```

Comment:
Here, you break encapsulation as you expose a C API data type. This looks wrong.